src/vnsw/agent/oper/interface.cc

```diff
@@ -579,15 +610,25 @@ void Interface::GetOsParams(Agent *agent) {
         return;
     }
 
-    if (!intf_luid_) {
-        intf_luid_ = GetInterfaceLuidFromName(name, type_);
+    /* Get interface's GUID. Should only happen on first call of `GetOsParams`. */
+    if (!os_guid_) {
+        NET_LUID net_luid = GetInterfaceLuidFromName(name, type_);
+        os_guid_ = GetInterfaceGuidFromLuid(net_luid);
     }
 
-    /* We assume that interface is UP */
-    os_oper_state_ = true;
-    os_index_ = GetInterfaceIndexFromLuid(*intf_luid_);
-    name_ = GetInterfaceNameFromLuid(*intf_luid_);
-    mac_ = GetMacAddressFromIndex(os_index_);
+    if (os_guid_) {
+        auto net_luid = GetInterfaceLuidFromGuid(*os_guid_);
+
+        os_index_ = GetInterfaceIndexFromLuid(net_luid);
+        name_ = GetInterfaceNameFromLuid(net_luid);
+        mac_ = GetMacAddressFromLuid(net_luid);
+
+        /* We assume that interface is UP */
+        os_oper_state_ = true;
+    } else {
+        LOG(ERROR, "interface GUID not set on " << name);
+        assert(0);
```

Comment:
Now this `else` is useless. When you change the return type of GetInterfaceGuidFromLuid it probably will be usefull.